### PR TITLE
fix Issue 3632 - modify float is float to do a bitwise compare

### PR DESCRIPTION
**Pull request by dkorpel**
Reboot of https://github.com/dlang/dmd/pull/7568

### CODE REVIEW

**Comment by dkorpel** [comment]
Currently fails on Windows:
```
test28_0.obj : fatal error LNK1179: invalid or corrupt file: duplicate COMDAT '_D4test__T3T46VdeNANZQld'
```

That's `double test.T46!(real.nan).T46`, relevant code:

```D
template T46(double v)
{
    double T46 = v;
}

void test46()
{
    double g = T46!(double.nan) + T46!(-double.nan);
}
```

**Comment by dkorpel** [comment]
Currently fails on Windows:
```
test28_0.obj : fatal error LNK1179: invalid or corrupt file: duplicate COMDAT '_D4test__T3T46VdeNANZQld'
```

That's `double test.T46!(real.nan).T46`, relevant code:

```D
template T46(double v)
{
    double T46 = v;
}

void test46()
{
    double g = T46!(double.nan) + T46!(-double.nan);
}
```

File: src/dmd/e2ir.d [hunk 1]
```diff
@@ -2024,7 +2024,11 @@ extern (C++) class ToElemVisitor : Visitor
             elem *es2 = toElem(ie.e2, irs);
             es2 = addressElem(es2, ie.e2.type);
             e = el_param(es1, es2);
-            elem *ecount = el_long(TYsize_t, t1.size());
+            elem *ecount;
+            if (t1.ty == TY.Tfloat80)
```
Comment (maxhaton) [inline src/dmd/e2ir.d:2028]: This could be a ternary + comment and link to issue appreciated since the glue code is horrible as it is..

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
cc @kinke does LDC still have that real padding issue?

**Comment by kinke** [comment]
The padding is still undefined, but `is` is restricted to the actual payload (I think I've fixed that quite some years ago...).

File: src/dmd/ctfeexpr.d [hunk 2]
```diff
@@ -1408,6 +1408,34 @@ bool ctfeIdentity(const ref Loc loc, EXP op, Expression e1, Expression e2)
         complex_t v2 = e2.toComplex();
         cmp = RealIdentical(creall(v1), creall(v2)) && RealIdentical(cimagl(v1), cimagl(v1));
     }
+    else if (e1.op == EXP.structLiteral || e2.op == EXP.structLiteral)
```
Comment (kinke) [inline src/dmd/ctfeexpr.d:1411]: What's all this struct literal business, does it have anything to do with floating-point identity?

Comment (dkorpel) [reply src/dmd/ctfeexpr.d]: Not sure, I did not originally write it, I rebased Iain's PR which rebased another PR from 2013. I removed it and modified `ctfeRawCmp` now.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**